app/controllers/spree/admin/base_controller.rb

```diff
@@ -33,7 +33,7 @@ def warn_invalid_order_cycles
 
       def model_class
         const_name = controller_name.classify
-        return "Spree::#{const_name}".constantize if Spree.const_defined?(const_name)
+        return Spree.const_get(const_name) if Spree.const_defined?(const_name)
```

Comment:
```diff
-        return "Spree::#{const_name}".constantize if Spree.const_defined?(const_name)
+        return Spree.const_get(const_name) if Spree.const_defined?(const_name)
```
A problem occurs when `const_name` = `Invoice`.

 the requests don't hit the right controller `Spree::Admin::InvoicesController`.
I'm not sure about the source of the problem, but here's the current behavior:

```ruby
3.0.3 :018 > Object.const_defined?("Spree::Invoice")
 => false 
3.0.3 :019 > Spree.const_defined?("Invoice") # From this, I expect Spree::Invoice to be defined
 => true 
3.0.3 :020 > "Spree::Invoice".constantize # < I can't explain this !
/.../activesupport-7.0.4.3/lib/active_support/inflector/methods.rb:280:in `const_get': uninitialized constant Spree::Invoice (NameError) 
```
I think it's better to use `Object.const_defined?`
```diff
-        return "Spree::#{const_name}".constantize if Spree.const_defined?(const_name)
+        return "Spree::#{const_name}".constantize if Object.const_defined?("Spree::#{const_name}")
```

Reply:
I found this topic  but I'm not it helps
https://community.openfoodnetwork.org/t/controllers-structure-a-cleanup-strategy/1994/4